src/neural/encoder.cc

```diff
@@ -182,13 +182,15 @@ InputPlanes EncodePositionForNN(
         // h-side (kingside) castling right.
         const auto& cast = board.castlings();
         result[kAuxPlaneBase + 0].mask =
-            ((cast.we_can_000() ? BoardSquare(ChessBoard::A1).as_board() : 0) |
-             (cast.they_can_000() ? BoardSquare(ChessBoard::A8).as_board() : 0))
-            << cast.queenside_rook();
+            ((cast.we_can_000() ? BoardSquare(ChessBoard::A1).as_board() : 0)
```

Comment:
This can be simplified a bit to:
```
(cast.we_can_000() ? BoardSquare(ChessBoard::A1 + cast.our_queenside_rook()).as_board() : 0) |
(cast.they_can_000() ? BoardSquare(ChessBoard::A8 + cast.their_queenside_rook()).as_board() : 0);
```